src/common/image.c

```diff
@@ -2304,7 +2304,7 @@ void dt_image_synch_all_xmp(const gchar *pathname)
       (dt_database_get(darktable.db),
        "SELECT id"
        " FROM main.images"
-       " WHERE film_id IN (SELECT id FROM main.film_rolls "
+       " WHERE film_id = (SELECT id FROM main.film_rolls "
```

Comment:
This is not equivalent.
there may be several filmrolls for a single folder (multiple imports for images a of single folder), right ?

Reply:
I don't think this is possible or maybe there is a feature that I have never seen. When you import a single image it is automatically placed into the folder corresponding to the directory the image is in. If you do that a second time for another image the second image will be placed into the same folder.

Reply:
Just checked. You are right. film_id remains the same. My mistake.

Then that's ok for this one.